gpii/node_modules/matchMakerFramework/src/MatchMakerFramework.js

```diff
@@ -103,6 +110,150 @@
     };
 
     /*
+	 * Returns a string indicating the matchmaker to select from configuration. Depends on the matchmaker payload.
+     */
+    gpii.matchMakerFramework.selectMatchMaker = function (that, matchMakerPayload) {
+        if (that.options.matchMakerDispatcherDisabled) {
+            return "flat"
+        }
+		var result = gpii.matchMakerFramework.GetPreferredMatchMaker(matchMakerPayload);
+		return (result) ? result : gpii.matchMakerFramework.GetCapableMatchMaker(matchMakerPayload);
+    };
+
+    /*
+	 * Returns if a matchmaker payload contains preferences for selecting a matchmaker.
+	 * Returns NULL if no matchmaker preference was found.
+     */
+	gpii.matchMakerFramework.GetPreferredMatchMaker = function (matchMakerPayload) {
+		// Rejection tests
+		if (matchMakerPayload.preferences === undefined || matchMakerPayload.preferences.contexts === undefined || matchMakerPayload.activeContexts === undefined) {
+			return null;
+		}
+		// Search for mm selection preferences
+		var contexts = matchMakerPayload.preferences.contexts;
+		var activeContexts = matchMakerPayload.activeContexts;
+		var foundPriority = -1;
+		var foundMatchMaker;
+		fluid.each(activeContexts, function (activeContext) {
+			if (contexts[activeContext] && contexts[activeContext].preferences) {
+				var prefs = contexts[activeContext].preferences;
+				var curPriority = 0;
+				if (contexts[activeContext].priority) {
+					curPriority = contexts[activeContext].priority;
+				}
+				if ((prefs["http://registry.gpii.net/common/matchMakerType"]) && (curPriority > foundPriority)) {
+					foundPriority = curPriority;
+					foundMatchMaker = prefs["http://registry.gpii.net/common/matchMakerType"];
+				}
+			}
+		});
+		return foundMatchMaker;
+    };
+
+    /*
+	 * Returns the Matchmaker that best matches to the playload based on its capabilities.
+	 * See MatchMakerCapabilities.json.
+     */
+	gpii.matchMakerFramework.GetCapableMatchMaker = function (matchMakerPayload) {
+		// Contexts
+		var contexts = matchMakerPayload.preferences.contexts;
+		// Find excluded matchmakers
+		var excludedMatchMakers = [];
+		var activeContexts = matchMakerPayload.activeContexts;
+		fluid.each(activeContexts, function (activeContext) {
+			if (contexts[activeContext] && contexts[activeContext].preferences) {
+				var prefs = contexts[activeContext].preferences;
+				if (prefs["http://registry.gpii.net/common/excludedMatchMakers"]) {
+					excludedMatchMakers = prefs["http://registry.gpii.net/common/excludedMatchMakers"];
+				}
+			}
+		});
+		// Try again block
+		if ( matchMakerPayload.tryAgain && matchMakerPayload.tryAgain.active) {
+			if (matchMakerPayload.tryAgain[excludedMatchMakers]) {
```

Comment:
This looks wrong... `excludedMatchMakers` is an array, so it doesn/t make much sense to use it as a key... Have you tested whether this works?